### PR TITLE
[C-2649] Refactor desktop collection header

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Refactors collection header elements to typescript, and isolates components into cleaner chunks.

#### TODO: Migrate CollectionHeader itself to ts

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2649-collection-header

File: packages/web/src/components/collection/desktop/PlayButton.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import { Button, ButtonType, IconPause, IconPlay } from '@audius/stems'
+import cn from 'classnames'
+
+import styles from './CollectionHeader.module.css'
+import { BUTTON_COLLAPSE_WIDTHS } from './utils'
+
+const messages = {
+  play: 'Play',
+  pause: 'PAuse'
```
Comment (Kyle-Shanks) [inline packages/web/src/components/collection/desktop/PlayButton.tsx:11]: PAuse

Comment (Kyle-Shanks) [reply packages/web/src/components/collection/desktop/PlayButton.tsx:11]: probably not a big deal if we transform to uppercase anyway

**Review by Kyle-Shanks — APPROVED** [review]

File: packages/web/src/components/collection/desktop/PlayButton.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/components/collection/desktop/PlayButton.tsx:11]: great catch

**Pull request merged**